packages/fetchai/skills/tac_control/behaviours.py

```diff
@@ -95,14 +97,40 @@ def teardown(self) -> None:
         self._unregister_tac()
         self._unregister_agent()
 
-    def _retry_failed_registration(self) -> None:
+    def retry_failed_registration(self) -> None:
         """Retry a failed registration."""
         if self.failed_registration_msg is not None:
             self._nb_retries += 1
             if self._nb_retries > self._max_soef_registration_retries:
                 self.context.is_active = False
                 return
 
+            if (
+                self.failed_registration_reason
+                == OefSearchMessage.OefErrorOperation.ALREADY_IN_LOBBY
+            ):
+                parameters = cast(Parameters, self.context.parameters)
+                self.context.logger.info(
+                    "Already in OEF lobby from previous registration attempt."
+                )
+                self.context.logger.info(
+                    f"Retrying registion in {parameters.registration_retry_interval} seconds."
+                )
+                sleep(parameters.registration_retry_interval)
```

Comment:
This is wrong. We can't sleep in behaviour/handler; messes up the async design. 

Instead of sleeping for some time before continuing, we should have a ticker behaviour that periodically (the period could be made to match the desired sleep duration) checks if a condition is met for moving forward with he code execution.